services/rds.py

```diff
@@ -1,95 +1,290 @@
 from models import RuleCheckResult
+import datetime
+from dateutil.tz import tzlocal
 import boto3
 
-
-# client = boto3.client("")
+client = boto3.client("rds")
+backup_client = boto3.client("backup")
+ec2_client = boto3.client("ec2")
 
 
 def aurora_last_backup_recovery_point_created():
+    compliant_resources = []
+    non_compliant_resources = []
+    clusters = client.describe_db_clusters()["DBClusters"]
+
+    for cluster in clusters:
+        recovery_points = backup_client.list_recovery_points_by_resource(ResourceArn=cluster["DBClusterArn"])[
+            "RecoveryPoints"
+        ]
+        recovery_point_creation_dates = sorted([i["CreationDate"] for i in recovery_points])
+        if datetime.datetime.now(tz=tzlocal()) - recovery_point_creation_dates[-1] < datetime.timedelta(days=1):
+            compliant_resources.append(cluster["DBClusterArn"])
+        else:
+            non_compliant_resources.append(cluster["DBClusterArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def aurora_mysql_backtracking_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    clusters = client.describe_db_clusters()["DBClusters"]
+
+    for cluster in clusters:
+        if cluster["Engine"] == "aurora-mysql" and cluster.get("EarliestBacktrackTime", None) == None:
+            non_compliant_resources.append(cluster["DBClusterArn"])
+        else:
+            compliant_resources.append(cluster["DBClusterArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def db_instance_backup_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    clusters = client.describe_db_clusters()["DBClusters"]
+
+    for cluster in clusters:
+        if cluster.get("BackupRetentionPeriod", None) != None:
+            compliant_resources.append(cluster["DBClusterArn"])
+        else:
+            non_compliant_resources.append(cluster["DBClusterArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def rds_cluster_auto_minor_version_upgrade_enable():
+    compliant_resources = []
+    non_compliant_resources = []
+    clusters = client.describe_db_clusters()["DBClusters"]
+
+    for cluster in clusters:
+        if cluster.get("AutoMinorVersionUpgrade", None) == True:
```

Comment:
DocDB에는 적용되지 않는 rule입니다.
```suggestion
        if cluster["Engine"] == "docdb" or cluster.get("AutoMinorVersionUpgrade"):
```